src/System.Console/src/System/IO/StdInStreamReader.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    /* This class is used by for reading from the stdin.
+     * It is designed to read stdin in raw mode for interpreting
+     * key press events and maintain its own buffer for the same.
+     * which is then used for all the Read operations
+     */
+    internal class StdInStreamReader : StreamReader
+    {
+        char[] unprocessedBufferToBeRead; // Buffer that might have already been read from stdin but not yet processed.
+        const int bytesToBeRead = 255; // No. of bytes to be read from the stream at a time.
+        int startIndex; // First unprocessed index in the buffer;
+        int endIndex; // Last unprocessed index in the buffer;
+
+        internal StdInStreamReader(Stream stream, Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize, bool leaveOpen) : base(stream, encoding, detectEncodingFromByteOrderMarks, bufferSize, leaveOpen)
```

Comment:
Do we ever construct one of these with varied values for leaveOpen or detectEncodingFromByteOrderMarks?  If not, they can be removed from this new ctor, and the constant values can be used when calling to base.